extensions/gitpod-web/package.json

```diff
@@ -516,25 +516,30 @@
   "scripts": {
     "postinstall": "yarn install:webview",
     "vscode:prepublish": "yarn build; yarn run compile --cwd ..",
-    "build": "yarn build:webview && yarn compile:ts",
+    "build": "yarn jest; yarn build:webview && yarn compile:ts",
     "watch": "npm-run-all -lp watch:webview watch:ts",
     "compile:ts": "tsc -p ./",
     "watch:ts": "tsc -watch -p ./",
     "install:webview": "yarn --cwd ./webview-ui --modules-folder node_modules",
     "watch:webview": "yarn --cwd ./webview-ui run dev",
-    "build:webview": "yarn --cwd ./webview-ui run build"
+    "build:webview": "yarn --cwd ./webview-ui run build",
+    "test": "jest"
```

Comment:
we should use mocha as vscode does, no need for a new test framework to learn :smile:

Reply:
I tried to setup with mocha, but failed 🙈

Reply:
I'll have another try with `mocha` setup

Reply:
@jeanp413  If we use `mocha`, 

- We need `ts-node/register` for `.ts` testing
- `ts-node` must install in current folder, otherwise, mocha will said `ts-node/register` not found
- After `nohoist` ts-node in current folder, mocha will say `gitpod-shared` not found

We can't move gitpod related module inside extension folder itself, otherwise their refer of `node_modules` will be different, and will cause an error like before [internal chat](https://gitpod.slack.com/archives/C01KGM9BH54/p1655831041154569?thread_ts=1655827615.556309&cid=C01KGM9BH54)

Maybe there have another ways, but I can't found it

Reply:
why do we need ts-node? `.test.ts` files should be compiled to js too

Reply:
At least for now we decided to use `jest`, it'll inside our build steps (yes, werft failed again 😅 will try clean slate later)

|Test in Werft Build| 
|-|
|![image](https://user-images.githubusercontent.com/20944364/175605778-a0a22865-dbd2-4728-9396-3ec4047fb312.png)|

| Test Coverage |
|-|
|![SCR-20220624-vun](https://user-images.githubusercontent.com/20944364/175605495-e1841e71-f31d-4650-9447-ec6307b7dbe1.png)|